fix(DOC-1940): clarify rpk topic consume keeps connection open (#1612)
redpanda-data/documentation: versioned_docs/version-23.2/get-started/config-rpk-profile.mdx

---
title: rpk Profiles
---

<head>
    <meta name="title" content="rpk Profiles | Redpanda Docs"/>
    <meta name="description" content="Use rpk profiles to simplify your development experience with multiple Redpanda clusters by saving and reusing configurations for different clusters."/>
    <link rel="canonical" href="https://docs.redpanda.com/docs/get-started/config-rpk-profile/" />
</head>

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

Use rpk profiles to simplify your development experience using `rpk` with multiple Redpanda clusters by saving and reusing configurations for different clusters.

## About rpk profiles

> An rpk profile contains a reusable configuration for a Redpanda cluster. When running `rpk`, you can create a profile, configure it for a cluster you're working with, save it, then use it when running an rpk command for the cluster. 

Reviewer: `save it` -- I think this is implied by `create` and `configure`.

wdyt of:
An rpk profile contains a reusable configuration for a Redpanda cluster. When running `rpk`, you can create a profile, configure it for a cluster you're working with, and then repeatably use it when running an rpk command for the cluster.

> You can create different profiles for different Redpanda clusters. For example, your local cluster, development cluster, and production cluster can each have their own profile, with all of their information stored in a single, local file. You set a unique name for each profile.

Reviewer: `single, local file` is too specific, I try to hide this implementation detail in rpk (what if we need more files later?)
wdyt of: "with all of their information managed locally by rpk" ?

> A profile saves the Redpanda node and cluster properties that rpk can already configure. For details, see [Specifying command properties](../intro-to-rpk#specifying-command-properties).

Reviewer: Not node and cluster properties, just rpk-specific-command properties.

> Most `rpk` commands can read configuration values from a profile. You pass a profile to an `rpk` command by setting the `--profile` flag. For example, the command `rpk topic produce dev-topic --profile dev` gets its configuration from the profile named `dev`.

Reviewer 2: My misunderstanding, thought I saw a command w/o `--profile` but they all actually do.

## Work with rpk profiles

The primary tasks for working with rpk profiles:

- Create one or more profiles.

> - Choose the in-use profile.

Reviewer: `Choose the profile to use`?

- Edit or set default values across all profiles and values for a single profile.
- Call an `rpk` command with a profile.
- Delete unused profile(s).

### Create profile

To create a new profile, run [`rpk profile create`](../../reference/rpk/rpk-profile/rpk-profile-create):

```bash
rpk profile create <profile-name> [flags]
```

> If no profile name is provided, the name `default` is used. After the profile is created, rpk switches to the newly created profile.

Reviewer: I removed the `default` name in the latest commit -- it was actually pretty confusing. Instead, a name is required _unless_ the person uses `--from-cloud`, where we will generate the name from the cloud cluster name (if no name is provided at the cli)

Reviewer 2: Will fix

An rpk profile can be generated from different sources:
- A `redpanda.yaml` file, using the `--from-redpanda` flag.
- A different rpk profile, using the `--from-profile` flag.
- A Redpanda Cloud cluster, using the `--from-cloud` flag.

Configuration can be set during creation with the `--set [key=value]` flag. To simplify configuration, the `--set` flag supports autocompletion of valid keys, suggesting key names based on their `-X` format. 

> :::important

Reviewer: maybe not _so_ important to have a full different CSS block for it, but this is worth nothing.

Reviewer 2: Changing to `:::note`

You should always use and set the `--description` flag to describe your profiles. The description is printed in the output of [`rpk profile list`](../../reference/rpk/rpk-profile/rpk-profile-list).
:::

> Created profiles are stored in an `rpk.yaml` file. By default, this file is created as `~/.config/rpk/rpk.yaml`. All profiles created by a developer are stored in the same `rpk.yaml` file.

Reviewer: It is `~/Library/Application Support/rpk/` on Mac, and somewhere else on windows -- i.e. the OS localized config dir.

Reviewer 3: %AppData% For Windows, but we don't really document the Windows usage.

All `rpk` commands can read configuration values from a profile. You pass a profile to an `rpk` command by setting the `--profile` flag. For example, the command `rpk topic produce dev-topic --profile dev` gets its configuration from the profile named `dev`.

## Work with rpk profiles

The primary tasks for working with rpk profiles:

- Create one or more profiles.
- Choose the profile to use.
- Edit or set default values across all profiles and values for a single profile.
- Call an `rpk` command with a profile.
- Delete unused profile(s).

### Create profile

To create a new profile, run [`rpk profile create`](../../reference/rpk/rpk-profile/rpk-profile-create):

```bash
rpk profile create <profile-name> [flags]
```

An rpk profile can be generated from different sources:
- A `redpanda.yaml` file, using the `--from-redpanda` flag.
- A different rpk profile, using the `--from-profile` flag.
- A Redpanda Cloud cluster, using the `--from-cloud` flag.

:::note
You must provide a profile name when creating a profile that isn't generated from a Redpanda Cloud cluster with the `--from-cloud` flag.
:::
 

> After the profile is created, rpk switches to the newly created profile.

Reviewer 4: ```suggestion
After the profile is created, rpk switches to the newly-created profile.
```

> Configuration can be set during creation with the `--set [key=value]` flag. To simplify configuration, the `--set` flag supports autocompletion of valid keys, suggesting key names based on their `-X` format. 

Reviewer 4: ```suggestion
You can specify the configuration during creation with the `--set [key=value]` flag. To simplify configuration, the `--set` flag supports autocompletion of valid keys, suggesting key names based on their `-X` format. 
```
